### PR TITLE
PB-496: External WMTS layer in 3d

### PR DESCRIPTION
**Pull request by ltshb**
TODO in separate PR:
- [ ] time series support for external WMS
- [ ] Proper time series update (see /PB-518)
[Test link](https://sys-map.dev.bgdi.ch/preview/bug-pb-496-external-wmts-3d/index.html)

### CODE REVIEW

File: src/api/layers/ExternalLayer.class.js [hunk 1]
```diff
@@ -148,5 +156,9 @@ export default class ExternalLayer extends AbstractLayer {
             this.availableProjections.push(WGS84)
         }
         this.getFeatureInfoCapability = getFeatureInfoCapability
+        this.currentYear = currentYear
```
Comment (pakb) [inline src/api/layers/ExternalLayer.class.js:159]: does it really make sense to store this value, if we do not react/update the time config whenever it changes?

File: src/modules/map/components/openlayers/OpenLayersExternalWMTSLayer.vue [hunk 2]
```diff
@@ -29,11 +30,39 @@ const { externalWmtsLayerConfig, parentLayerOpacity, zIndex } = toRefs(props)
...
+const previewYear = computed(() => store.state.layers.previewYear)
+const isTimeSliderActive = computed(() => store.state.ui.isTimeSliderActive)
 
 // extracting useful info from what we've linked so far
 const layerId = computed(() => externalWmtsLayerConfig.value.id)
 const opacity = computed(() => parentLayerOpacity.value ?? externalWmtsLayerConfig.value.opacity)
-const options = computed(() => externalWmtsLayerConfig.value.options)
+const options = computed(() => {
+    const options = { ...externalWmtsLayerConfig.value.options }
+    options.dimensions = { ...options.dimensions }
```
Comment (pakb) [inline src/modules/map/components/openlayers/OpenLayersExternalWMTSLayer.vue:41]: Might make sense that we add some cloneDeep functions from Lodash or other library for such cases

**Review by pakb — APPROVED** [review]
That was a lot to add to achieve the wanted result :open_mouth:

File: src/api/layers/ExternalLayer.class.js (hunk 1 above)
Comment (ltshb) [reply src/api/layers/ExternalLayer.class.js:159]: It is used by the external layer plugin once the capabilities has been loaded, so it is required

**Comment by ltshb** [comment]
> That was a lot to add to achieve the wanted result 😮

I went a bit further by enabling time layer in external import as well 😉

**Pull request merged**